app/controllers/search_controller.rb

```diff
@@ -5,7 +5,11 @@ def index
     if @form.valid?
       @result = FirmRepository.new.search(@form.to_query, page: page)
     else
-      render 'landing_page/show'
+      if params[:origin] == 'results'
```

Comment:
So, I know this is a terrible approach, wondering if we have any other ideas for tackling the problem that this solves?